crates/services/p2p/src/cached_view.rs

```diff
@@ -0,0 +1,320 @@
+use crate::ports::P2pDb;
+use dashmap::DashMap;
+use fuel_core_metrics::p2p_metrics::{
+    increment_p2p_req_res_cache_hits,
+    increment_p2p_req_res_cache_misses,
+};
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    blockchain::SealedBlockHeader,
+    services::p2p::Transactions,
+};
+use std::ops::Range;
+type BlockHeight = u32;
+
+pub(super) struct CachedView {
+    sealed_block_headers: DashMap<BlockHeight, SealedBlockHeader>,
+    transactions_on_blocks: DashMap<BlockHeight, Transactions>,
+    metrics: bool,
+}
+
+impl CachedView {
+    pub fn new(metrics: bool) -> Self {
+        Self {
+            sealed_block_headers: DashMap::new(),
+            transactions_on_blocks: DashMap::new(),
+            metrics,
+        }
+    }
+
+    pub(super) fn clear(&self) {
```

Comment:
Maybe we could use some LRU cache instead of wiping the whole cache every 10 seconds? This is just a thought since the current approach should also work.

Reply:
Yeah, in the future we can use an LRU :) we discussed it somewhere above too